src/main/java/com/codeofus/rent_a_park/services/PersonService.java

```diff
@@ -0,0 +1,94 @@
+import com.codeofus.rent_a_park.models.Spot;
+import com.codeofus.rent_a_park.repositories.PersonRepository;
+import lombok.AccessLevel;
+import lombok.RequiredArgsConstructor;
+import lombok.experimental.FieldDefaults;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Optional;
+
+@FieldDefaults(level = AccessLevel.PRIVATE, makeFinal = true)
+@RequiredArgsConstructor
+@Transactional(readOnly = true)
+@Service
+public class PersonService {
+
+    private PersonRepository personRepository;
+
+    private ParkingMapper mapper;
+
+    public List<PersonDto> getAll() {
+        return personRepository.findAll().stream().map(mapper::personToDto).toList();
+    }
+
+    @Transactional
+    public Person addNewPerson(PersonDto personDto) {
+        return personRepository.save(mapper.toPerson(personDto));
+    }
+
+    @Transactional
+    public Optional<PersonDto> updatePerson(PersonDto person) {
+        return Optional
+                .of(personRepository.findById(person.getId()))
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .map(
+                        user -> {
+                            user.setFirstName(person.getFirstName());
+                            user.setLastName(person.getLastName());
+                            user.setRegistration(person.getRegistration());
+                            user = personRepository.save(user);
+                            return user;
+                        }
+                )
+                .map(mapper::personToDto);
+    }
+
+    @Transactional
+    public void deletePerson(Integer personId) {
+        personRepository
+                .findOneById(personId)
+                .ifPresent(
+                        personRepository::delete
+                );
+    }
+
+    @Transactional
+    public void reserveParkingSpot(Spot spot, Person parker) {
+        List<Spot> parkingSpots = parker.getParkingSpots();
+        parkingSpots.add(spot);
+        if (personRepository.findAll().contains(parker)) {
+            personRepository.delete(parker);
+        }
+        parker.setParkingSpots(parkingSpots);
+        personRepository.save(parker);
+    }
+
+    @Transactional
+    public void addParkingSpot(Spot spot, Person renter) {
+        List<Spot> rentedSpots = renter.getRentedSpots();
+        rentedSpots.add(spot);
+        if (personRepository.findAll().contains(renter)) {
+            personRepository.delete(renter);
+        }
+        renter.setRentedSpots(rentedSpots);
+        personRepository.save(renter);
+    }
+
+    @Transactional
+    public void cancelReservation(Spot spot, Person parker) {
+        List<Spot> parkingSpots = parker.getParkingSpots();
+        parkingSpots.remove(spot);
+        personRepository.delete(parker);
+        parker.setParkingSpots(parkingSpots);
+        personRepository.save(parker);
```

Comment:
is .save needed since we're inside the @Transactional. Can you write a test to check that?